jgiven-core/src/main/java/com/tngtech/jgiven/annotation/FillerWord.java

```diff
@@ -0,0 +1,26 @@
+package com.tngtech.jgiven.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marks a method which will be used as a preamble to the next non-filler stage method.
+ * In reports, filler words inherit the alignment of the stage method that follows.
+ * Examples of filler words that may be used are the, some, for_a.
+ * domain specific filler words can be created for specific use cases.
+ */
+@Documented
+@Retention( RetentionPolicy.RUNTIME )
+@Target( ElementType.METHOD )
+public @interface FillerWord {
+
+    /**
+     * In reports, filler words marked with affix will be appended to the previous filler word
+     * in the sentence without whitespace.
+     */
+    boolean affix() default false;
```

Comment:
fine by me. Maybe `affixToWord` for some additional clarity

Reply:
After a bit of playing around tests I opted for the use of affix "noun" rather than the "verb". So a filler word can be used as an affix rather than being "affixed" to a sentence. More than happy to switch it if the noun doesn't quite feel right.

Reply:
Maybe that meaning just doesn't come that naturally to me...If we see it as a noun then we could name the field `isAffix` just to highlight that it is a noun

Reply:
The more I think about it the less I like "afffix"! I'll have a think over the weekend how I can better convey the behaviour. We may arrive back at "whitespace".

Reply:
I'm not completely sold on "affix" either, because personally I would have to search online to get the exact meaning :joy: 
Also if you're very strict, then "an addition to the base form or stem of a word in order to **modify its meaning** or **create a new word**" is not satisfied for the use case you're thinking about, is it? :thinking: Adding punctuation neither modifies the meaning nor creates a new word, so it kind of hides the intention it was originally created for, right?
`whitespace` is technical enough to not have this, but with `whitespace` my problem is that it does not tell you "before or after", does it? :thinking: I might also think then "either it has a whitespace before and after or neither before nor after" which is not true for punctuation either, right?
Don't have a great idea here, but something like `whitespaceBefore` or `joinToLastWord` would at least make it clear that it only refers to the word before this filler word. Or you just support the two use cases of normal words and punctuation and have an attribute like `punctuation`. Which would also match to names on `Word` like `addWord` and `addPunctuation`.

Reply:
I agree, I have never heard the word affix before, so maybe not optimal for non-native speakers ;-). Regarding that it is unclear where the space is added, I think this can be clarified in the JavaDoc comment. As space is the default, we could also just called it `noSpace`.

Reply:
Another alternative would be to have an attribute `prefix` which would be a String instead of a boolean. By default it would be `" "` and can be changed to `""`. But it could also be anything else, like `"-"` for example. I am not sure, however, whether there is any useful use-case for that.

Reply:
My current thoughts are along the lines of `whitespaceBefore` and `whitespaceAfter` allowing you to nicely use parentheses if you so desire.

Reply:
Or `joinToPreviousWord` and `joinToNextWord`.

Reply:
`affix` has been removed. We now have `joinToPreviousWord` and `joinToNextWord` implemented. Supporting both obviously add a bit more complexity but I think the having the ability to join to both previous and next words is a nice addition. Again, all feedback welcomed.

Reply:
I'll have a look at it first thing tomorrow morning